File: Leetcode/1. Two Sum/java/twoSum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        // 初始化 hash 表
        HashMap<Integer, Integer> hashableNums = new HashMap<>();

        // 存储数据
        for (int i = 0; i < nums.length; i++) {
            // 查找是否已加入 hash 表
            if (hashableNums.containsKey(target - nums[i])) {
                // 如果找到结果，直接返回
                return new int[]{hashableNums.get(target - nums[i]), i};
            }
            // 没有找到结果，加入 hash 表
            hashableNums.put(nums[i], i);
        }
        return new int[]{};
    }
}

File: Leetcode/1. Two Sum/javascript/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    // 初始化 hash 表
    hashableNums = {}

    // 存储数据
    for (var i = 0; i < nums.length; i++) {
        // 查找是否已加入 hash 表
        if (target - nums[i] in hashableNums) {
            // 如果找到结果，直接返回
            return [hashableNums[target - nums[i]], i]
        }

        // 没有找到结果，加入 hash 表
        hashableNums[nums[i]] = i
    }

    return []
}

module.exports = {
    twoSum
}